### PR TITLE
Add deep Q-learning utilities for board game

### PR DESCRIPTION
**Pull request by evanfwelch**
## Summary
- add PyTorch-based utilities for encoding board state and actions
- provide replay buffer, dataset iterator, reward helper, and DQN agent with training updates
- expose new utilities and depend on torch

## Testing
- `PYTHONPATH=src pytest -q`

------
https://chatgpt.com/codex/tasks/task_e_689a161ce26083268bdab23dfd04bf0f

### CODE REVIEW

**Review by evanfwelch — COMMENTED** [review]
LGTM
